### ISSUE TITLE
Message export failing

### ISSUE THREAD

**Issue by vgeddes** [praekeltfoundation/vumi-go#702]
I'm unable to export messages for the following conversation: 

Name: GFM Nigeria
Conversation ID: d917d8f8354a4efb92ca7148174704ae
Account: vincent@praekeltconsulting.com

I looked through the celery logs, and found the following error which corresponds to my export task:

```
[2013-09-10 00:06:31,482: ERROR/MainProcess] Task go.conversation.tasks.export_conversation_messages[c586a115-97ac-4644-9a49-145b9c980de7] raised exception: WorkerLostError('Worker exited prematurely.',)
Traceback (most recent call last):
  File "/mnt/praekelt/vumi-go/ve/local/lib/python2.7/site-packages/celery/concurrency/processes/pool.py", line 634, in _join_exited_workers
    raise WorkerLostError("Worker exited prematurely.")
WorkerLostError: Worker exited prematurely.
```

The conversation has lots of activity, so we could be hitting some kind of imposed resource ceiling.

**Comment by vgeddes** [comment]
There's no logging in `https://github.com/praekelt/vumi-go/blob/develop/go/conversation/tasks.py`. At the very least, I'd hook-in the Sentry plugin for celery, which makes sure that all uncaught exceptions get reported to Sentry. See https://github.com/praekelt/gfm-ussd-service/blob/develop/figo/tasks/tasks.py#L45 for an example

Otherwise, perhaps the entity that kicked off the task had set a timeout parameter or something.

**Comment by smn** [comment]
Odd, thanks for reporting. It'd probably be a good idea to let the MessageStore API make the messages available as a streaming download.

**Comment by smn** [comment]
Looking at https://github.com/celery/celery/issues/1406 I'm guessing it's being OOMd because everything's being collected & gzipped in memory.

**Comment by vgeddes** [comment]
Yup, you're right. I looked at syslog, and there was indeed an OOM condition:

```
Sep 10 14:31:55 ip-10-64-135-190 kernel: [31492559.375660] Out of memory: Kill process 24647 (django-admin.py) score 475 or sacrifice child
Sep 10 14:31:55 ip-10-64-135-190 kernel: [31492559.375675] Killed process 24647 (django-admin.py) total-vm:4049820kB, anon-rss:3619708kB, file-rss:52kB
```

I'm keen on streaming the dump to S3, and then sending the link to the user via email.

**Comment by smn** [comment]
I think we can stream the download using twisted.web straight from Riak I think? It would make the whole thing operationally simpler and the current Go API allows for auth session sharing between Twisted and Django which should make this a relatively straight forward exercise.

The reason this wasn't implemented as such earlier is because we didn't have all the infrastructure in place back then (which we currently do have now).

**Comment by smn** [comment]
@hodgestar what are your thoughts on this?

**Comment by hodgestar** [comment]
The plan is to include this in the Go API:
- The web frontend will include a link to click on that connects to the Go API with correct authentication details.
- Go API will stream down the messages piece by piece.

This work can be done when #668 is being worked on.

**Comment by smn** [comment]
I think #668 is more work than we are currently anticipating since contacts can be accessed from pretty much everywhere in Vumi Go.

I'd propose to keep this separate and possibly tackled before #668 as currently anyone with a remotely successful project is unable to export their messages because stuff is getting OOMd, resulting in devs getting prodded to try & do this manually (which is roughly the same amount of work as implementing this ticket).

**Comment by hodgestar** [comment]
This will be addressed by #727.

**Comment by hodgestar** [comment]
Closing in favour of #727.
